controllers/rsyncnodedatamovement_controller.go

```diff
@@ -106,63 +161,101 @@ func (r *RsyncNodeDataMovementReconciler) Reconcile(ctx context.Context, req ctr
+			cmd = exec.CommandContext(ctxCancel, "sh", "-c", rsyncNode.Spec.Simulate)
+		} else {
+			cmd = exec.CommandContext(ctxCancel, "rsync", arguments...)
+		}
 
-	arguments = append(arguments, "--archive")
-	arguments = append(arguments, source)
-	arguments = append(arguments, destination)
+		// Set the appropriate permissions
+		if rsyncNode.Spec.UserId != uint32(os.Getuid()) || rsyncNode.Spec.GroupId != uint32(os.Getgid()) {
+			cmd.SysProcAttr = &syscall.SysProcAttr{
+				Credential: &syscall.Credential{
+					Uid: rsyncNode.Spec.UserId,
+					Gid: rsyncNode.Spec.GroupId,
+				},
+			}
+		}
+		log.Info("Rsync credential", "uid", rsyncNode.Spec.UserId, "gid", rsyncNode.Spec.GroupId)
 
-	cmd := exec.CommandContext(ctx, "rsync", arguments...)
+		// Run the command and wait for completion
+		log.Info("Executing rsync command", "source", source, "destination", destination)
+		out, cmdErr := cmd.CombinedOutput()
 
-	if rsyncNode.Spec.UserId != uint32(os.Getuid()) || rsyncNode.Spec.GroupId != uint32(os.Getgid()) {
-		cmd.SysProcAttr = &syscall.SysProcAttr{
-			Credential: &syscall.Credential{
-				Uid: rsyncNode.Spec.UserId,
-				Gid: rsyncNode.Spec.GroupId,
-			},
+		// Log output based on result
+		if errors.Is(ctxCancel.Err(), context.Canceled) {
+			log.Info("Rsync cancelled", "error", cmdErr, "output", string(out))
+		} else if cmdErr != nil {
+			log.Info("Rsync failure", "error", cmdErr, "output", string(out))
+		} else {
+			log.Info("rsync completed", "output", string(out))
 		}
-	}
-	log.Info("Rsync credential", "uid", rsyncNode.Spec.UserId, "gid", rsyncNode.Spec.GroupId)
 
-	out, err := cmd.Output()
+		// Record the completion info in the resource
+		rsyncNode.Status.EndTime = metav1.NowMicro()
+		rsyncNode.Status.State = nnfv1alpha1.DataMovementConditionTypeFinished
 
-	if err != nil {
-		if exitErr, ok := err.(*exec.ExitError); ok {
-			log.Info("Rsync failure", "error", string(exitErr.Stderr))
+		if errors.Is(ctxCancel.Err(), context.Canceled) {
+			rsyncNode.Status.Status = nnfv1alpha1.DataMovementConditionReasonCancelled
+		} else if cmdErr != nil {
+			rsyncNode.Status.Status = nnfv1alpha1.DataMovementConditionReasonFailed
+			rsyncNode.Status.Message = cmdErr.Error()
 		} else {
-			log.Info("Rsync failure", "error", err, "output", string(out))
+			rsyncNode.Status.Status = nnfv1alpha1.DataMovementConditionReasonSuccess
 		}
-	} else {
-		log.Info("rsync completed", "output", string(out))
-	}
-
-	// Record the completion status.
-	rsyncNode.Status.EndTime = metav1.NowMicro()
-	rsyncNode.Status.State = nnfv1alpha1.DataMovementConditionTypeFinished
 
-	if err != nil {
-		rsyncNode.Status.Status = nnfv1alpha1.DataMovementConditionReasonFailed
-		rsyncNode.Status.Message = err.Error()
-	} else {
-		rsyncNode.Status.Status = nnfv1alpha1.DataMovementConditionReasonSuccess
-	}
+		// Update the completion status and retry on conflicts
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
You have to refresh your version of the object as you do the loop over the Status.Update.

Reply:
If you get a conflict, that means the version of your resource is stale.  This is why you do another Get(), to refresh your version of the resource.

If you're not going to do the Get inside the RetryOnConflict, then stop using it to wrap your Status.Update.